### DISCUSSION TITLE
Comparison to ESP-AT firmware?

### DISCUSSION THREAD

**Discussion by rgov** [bozimmerman/Zimodem#110] [Q&A]
Espressif publishes a proprietary firmware called [ESP-AT](https://github.com/espressif/esp-at). Being actually open source and Apache-licensed makes Zimodem an attractive alternative. Do you know other major differences in the project features or goals?

For example: Zimodem does not appear to support TLS connections. ESP-AT is unlikely to gain support for IRC.

One of my dislikes of ESP-AT is that the modem responses are fairly inconsistent in formatting, which makes the software driver of the modem fairly complicated. I haven't evaluated Zimodem for this but I'm curious if any particular attention has been paid to consistency.

Other projects: [This comment](https://github.com/bozimmerman/Zimodem/discussions/99#discussioncomment-5056286) compares Zimodem to WiModem (proprietary). There's also [WifiModem](https://github.com/dhansel/WifiModem) by David Hansel.

**Comment by markaudacity** [comment]
Just found out the WifiModem project allows one to configure Telnet capabilities, which my WiModem232 does not, which is frustrating because I primarily use it with a Tandy M102, which only has a 40x8 LCD and has very slow LCD writes, both of which result in it being very difficult to use some BBSes. Either I have to wait through a (for the M102) huge ANSI art splash page I can't even read, or the connection times out while the modem is sending data to the M102.

Telnet can negotiate many of these limitations, which would make the whole shebang a lot more usable for me.
